### PR TITLE
Add keepalived template

### PR DESCRIPTION
**Pull request by mrbobbytables**
This PR adds a keepalived template to the catalog that we have been using internally. When deployed it enables VRRP based IP failover between nodes and is useful for edge or ingress hosts.

### CODE REVIEW

File: templates/keepalived/0/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      KEEPALIVED_AUTOCONF: true
+      KEEPALIVED_STATE: MASTER
+      KEEPALIVED_INTERFACE: ${interface}
+      KEEPALIVED_VIRTUAL_ROUTER_ID: ${router_id}
+      KEEPALIVED_UNICAST_SRC_IP: ${master_ip}
+      KEEPALIVED_UNICAST_PEER_0: ${backup_ip}
+      KEEPALIVED_TRACK_INTERFACE_1: ${interface}
+      KEEPALIVED_VIRTUAL_IPADDRESS_1: "\"${virtual_ip}\""
+    labels:
+      io.rancher.scheduler.affinity:host_label: ${host_label}=${master_label}
```
Comment (rawmind0) [inline templates/keepalived/0/docker-compose.yml]: What do you think to add optional sysctl update, with go templating?? 
Rename docker-compose.yml to docker-compose.yml.tpl and add
```
+    {{- if eq .Values.UPDATE_SYSCTL "true"}}
+      io.rancher.sidekicks: keepalived-sysctl
+    {{- end}}
```

File: templates/keepalived/0/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      KEEPALIVED_AUTOCONF: true
+      KEEPALIVED_STATE: BACKUP
+      KEEPALIVED_INTERFACE: ${interface}
+      KEEPALIVED_VIRTUAL_ROUTER_ID: ${router_id}
+      KEEPALIVED_UNICAST_SRC_IP: ${backup_ip}
+      KEEPALIVED_UNICAST_PEER_0: ${master_ip}
+      KEEPALIVED_TRACK_INTERFACE_1: ${interface}
+      KEEPALIVED_VIRTUAL_IPADDRESS_1: "\"${virtual_ip}\""
+    labels:
+      io.rancher.scheduler.affinity:host_label: ${host_label}=${backup_label}
```
Comment (rawmind0) [inline templates/keepalived/0/docker-compose.yml]: What do you think to add optional sysctl update, with go templating?? 
Rename docker-compose.yml to docker-compose.yml.tpl and add
```
+{{- if eq .Values.UPDATE_SYSCTL "true" }}
+      io.rancher.sidekicks: keepalived-sysctl
+  keepalived-sysctl:
+    labels:
+      io.rancher.container.start_once: true
+    network_mode: none  
+    image: rawmind/alpine-sysctl:0.1-1
+    privileged: true
+    environment: 
+      - "SYSCTL_KEY=net.ipv4.ip_nonlocal_bind" 
+      - "SYSCTL_VALUE=1"
+{{- end}}
```

File: templates/keepalived/0/rancher-compose.yml [hunk 3]
```diff
@@ -0,0 +1,52 @@
+.catalog:
+  name: "keepalived"
+  version: v1.0.1
+  description: "Keepalived VRRP based HA service."
+  minimum_rancher_version: v0.46.0
+  uuid: keepalived-1
+  questions:
```
Comment (rawmind0) [inline templates/keepalived/0/rancher-compose.yml]: UPDATE_SYSCTLl variable definition
```
+    - variable: "UPDATE_SYSCTL"
+      label: "Update host sysctl:"
+      description: |
+        Set true to update sysctl.
+        WARN: If set true, sysctl key net.ipv4.ip_nonlocal_bind will be set to 1.
+      default: false
+      required: true
+      type: enum
+      options:
+        - false
+        - true
```

**Comment by rawmind0** [comment]
Hi @mrbobbytables ... thanks so much for you PR... keepalived is a nice package to have in catalog. 

I propose you to add an optional sidekick to set the sysctl key net.ipv4.ip_nonlocal_bind to 1. Please, take a look at the change request.

**Comment by mrbobbytables** [comment]
I like that idea :) I'll get the changes in there today.

**Comment by mrbobbytables** [comment]
@rawmind0 Okay, all changes in and tested locally . :)

**Comment by rawmind0** [comment]
Cool @mrbobbytables.. :)

LGTM...

**Pull request merged**